src/runtime/audience-action-flow-test.js

```diff
@@ -1010,6 +1046,105 @@ describes.realWin('AudienceActionFlow', (env) => {
     await tick(10);
   });
 
+  it(`handles a SurveyDataTransferRequest with successful PPS storage in empty localStorage`, async () => {
+    const audienceActionFlow = new AudienceActionFlow(runtime, {
+      action: 'TYPE_REWARDED_SURVEY',
+      configurationId: 'configId',
+      onCancel: onCancelSpy,
+      autoPromptType: AutoPromptType.CONTRIBUTION,
+    });
+    activitiesMock.expects('openIframe').resolves(port);
+
+    const newIabTaxonomyMap = {
+      [Constants.PPS_AUDIENCE_TAXONOMY_KEY]: {values: ['1', '2']},
+    };
+    storageMock
+      .expects('set')
+      .withExactArgs('ppstaxonomies', JSON.stringify(newIabTaxonomyMap), true)
+      .once();
+
+    await audienceActionFlow.start();
+    const activityIframeViewMock = sandbox.mock(
+      audienceActionFlow.activityIframeView_
+    );
+    activityIframeViewMock.expects('execute').once();
+
+    const messageCallback =
+      messageMap[TEST_SURVEYDATATRANSFERREQUEST_WITHPPS.label()];
+    messageCallback(TEST_SURVEYDATATRANSFERREQUEST_WITHPPS);
+
+    await tick(10);
+
+    storageMock.verify();
+    activityIframeViewMock.verify();
+  });
+
+  it(`handles a SurveyDataTransferRequest with successful PPS storage in populated localStorage`, async () => {
+    const audienceActionFlow = new AudienceActionFlow(runtime, {
+      action: 'TYPE_REWARDED_SURVEY',
+      configurationId: 'configId',
+      onCancel: onCancelSpy,
+      autoPromptType: AutoPromptType.CONTRIBUTION,
+    });
+    activitiesMock.expects('openIframe').resolves(port);
+
+    const existingIabTaxonomyMap = {
+      [Constants.PPS_AUDIENCE_TAXONOMY_KEY]: {values: ['3', '4']},
```

Comment:
Can you add a value that's not in new to existing so that the test covers adding (de-duped) values to existing one?

Reply:
To be more specific with an example, I meant something like
- existing : 3, 4, 5
- new: 1, 2, 3
- expect: 1, 2, 3, 4, 5